tools/Customization/DevHome.FileExplorerSourceControlIntegration/Services/FileExplorerIntegrationUserSettings.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using DevHome.Common.Models;
+using DevHome.Common.Services;
+using Serilog;
+
+namespace DevHome.FileExplorerSourceControlIntegration.Services;
+
+public class FileExplorerIntegrationUserSettings
+{
+    private readonly LocalSettingsService? _localSettingsService;
+    private readonly Serilog.ILogger _log = Log.ForContext("SourceContext", nameof(FileExplorerIntegrationUserSettings));
+
+    public FileExplorerIntegrationUserSettings()
+    {
+        try
+        {
+            FileService fileService = new FileService();
+            var options = new Microsoft.Extensions.Options.OptionsWrapper<LocalSettingsOptions>(
+            new LocalSettingsOptions
+            {
+                ApplicationDataFolder = "DevHome/ApplicationData",
+                LocalSettingsFile = "LocalSettings.json",
+            });
+
+            // The LocalSettingsService has to be used and initialized from within the
+            // File Explorer Source Control Integration COM Server
+            _localSettingsService = new LocalSettingsService(fileService, options);
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, "Failed to get LocalSettingsService");
+        }
+    }
+
+    public bool IsFileExplorerVersionControlEnabled()
+    {
+        if (_localSettingsService != null && _localSettingsService.HasSettingAsync("VersionControlIntegration").Result)
+        {
+            return _localSettingsService.ReadSettingAsync<bool>("VersionControlIntegration").Result;
+        }
+
+        // DisabledByDefault
+        return false;
```

Comment:
If the switch is enabled by default will the return value need to change?  If it does, comment should be added to let users know about this relationship.